crashlogging/src/main/java/com/automattic/android/tracks/crashlogging/internal/ApplicationInfoProvider.kt

```diff
@@ -0,0 +1,14 @@
+package com.automattic.android.tracks.crashlogging.internal
+
+import android.content.Context
+import android.content.pm.ApplicationInfo
+
+internal interface ApplicationInfoProvider {
+    val debuggable: Boolean
+}
+
+internal class AndroidApplicationInfoProvider(context: Context) : ApplicationInfoProvider {
+    override val debuggable: Boolean = context.applicationInfo.let {
+        it.flags and ApplicationInfo.FLAG_DEBUGGABLE != 0
+    }
+}
```

Comment:
It's not really possible to test without setting up Robolectric or instrumantation tests. I think it's too much effort for this in this library, but I've validated this behavior on Day One Android: the app crashes in debug but not in release if reporting without initialization:

![Screenshot 2024-04-24 at 15 58 18](https://github.com/Automattic/Automattic-Tracks-Android/assets/5845095/bfa8ae02-b886-4cbf-9f33-7929051069cb)